state/execution.go

```diff
@@ -239,6 +249,19 @@ func (blockExec *BlockExecutor) Commit(
 		"app_hash", fmt.Sprintf("%X", res.Data),
 	)
 
+	if blockExec.sidecar != nil {
+		// update sidecar mempool
+		err = blockExec.sidecar.Update(
+			block.Height,
+			block.Txs,
+			deliverTxResponses,
+		)
+		if err != nil {
```

Comment:
Check when the sidecar.Update returns error. This could mess up consensus.

Reply:
UPDATE: https://github.com/angbrav/mev-tendermint/pull/1#discussion_r1120457229

Reply:
[Issue in collab repo.](https://github.com/informalsystems/audit-skip/issues/9)

Reply:
Fixed in https://github.com/skip-mev/mev-tendermint/pull/126